python/pyarrow/src/arrow/python/python_to_arrow.cc

```diff
@@ -836,11 +848,14 @@ class PyListConverter : public ListConverter<T, PyConverter, PyConverterTrait> {
 
   Status AppendSequence(PyObject* value) {
     int64_t size = static_cast<int64_t>(PySequence_Size(value));
+    RETURN_NOT_OK(AppendTo(this->list_type_, size));
     RETURN_NOT_OK(this->list_builder_->ValidateOverflow(size));
     return this->value_converter_->Extend(value, size);
   }
 
   Status AppendIterable(PyObject* value) {
+    auto size = static_cast<int64_t>(PyObject_Size(value));
```

Comment:
I suppose at the moment we get here, we know for sure that the object has a size? (eg a generator is iterable, but has no length. We do support generators as the main value passed to `pa.array(..)` (through `ConvertToSequenceAndInferSize`), but not nested in a list array, I think)

Reply:
At the moment, `AppendIterable()` is only called for Sets and Dictionary values views. I suppose generators would be routed through `AppendSequence()` instead? I did give it a try just to be safe:

```
>>> def gen():
...     yield [0]
...     yield [1]
...     yield [2]
...
>>> pa.array(gen(), type=pa.list_view(pa.int32()))
<pyarrow.lib.ListViewArray object at 0x12c96e6e0>
[
  [
    0
  ],
  [
    1
  ],
  [
    2
  ]
]
```